cmd/thanos/query.go

```diff
@@ -404,3 +424,19 @@ func removeDuplicateStoreSpecs(logger log.Logger, duplicatedStores prometheus.Co
 	}
 	return deduplicated
 }
+
+// Duplicate returns a duplicate string in the given string slice
+// or empty string if none was found.
+func duplicate(ss []string) string {
+	set := map[string]struct{}{}
+
+	for _, s := range []string(ss) {
+		if _, ok := set[s]; ok {
```

Comment:
what if multiple duplicates in string slice?

Reply:
I have to admit I took the existing code mechanically, but the user will only be informed about the first duplicate. As far as I can we are retaining the existing mechanism. Happy to enhance this here though, wdyt?

Reply:
we can rename it to `firstDuplicate` and LGTM :D